Assets/Source/Components/Player/PlayerBehavior.cs

```diff
@@ -185,45 +276,35 @@ private void UpdateExternalVelocity()
             if (externalVelocity.y.IsWithin(STABILIZATION_RATE, 0f)) { externalVelocity = externalVelocity.Copy(y: 0f); }
         }
 
-        public void ReactToHit(Collision2D collision, int baseDamage, float partDamageMultiplier)
+        private IEnumerator ReduceHealth(int damage)
         {
-            string collisionName = collision.otherCollider.gameObject.name;
-            if (!collisionName.Equals(bulletPrefab.name) && !isInvulnerable)
-            {
-                actorBehavior.Health -= baseDamage;
+            int framesToFinish = 35; //should probly get this from the animation somehow? the animation is 35 frames rn
+            float startingPercent = (float)actorBehavior.Health / actorBehavior.MaxHealth;
 
-                // Warn player if health is less than 10%
-                if (actorBehavior.Health > 0 && ((float)actorBehavior.Health / actorBehavior.MaxHealth) < 0.1f)
-                {
-                    Warning("Low Health");
-                }
+            actorBehavior.Health -= damage;
 
-                // todo: Once we have enemies, we should add a weight value to them which will affect the hard coded values below
+            float endingPercent = (actorBehavior.Health <= 0) ? 0.0f : (float)actorBehavior.Health / actorBehavior.MaxHealth;
+            float percentPerFrame = (startingPercent - endingPercent) / framesToFinish;
 
-                // Hit from the left side
-                if (collision.otherCollider.transform.position.x < transform.position.x)
-                {
-                    externalVelocity = externalVelocity.Copy(x: externalVelocity.x + 1f);
-                    cameraEffector.Trigger_Impact_Left();
-                }
-                // Hit from the right side
-                else
-                {
-                    externalVelocity = externalVelocity.Copy(x: externalVelocity.x - 1f);
-                    cameraEffector.Trigger_Impact_Right();
-                }
+            int currentFrame = 0;
+            while(currentFrame != framesToFinish)
+            {
+                healthImage.fillAmount -= percentPerFrame;
+                currentFrame++;
+                yield return null;
+            }
 
-                // Hit from the ass
-                if (collision.otherCollider.transform.position.y < transform.position.y)
-                {
-                    externalVelocity = externalVelocity.Copy(y: externalVelocity.x + 1f);
-                }
-                // Hit from the right side
-                else
-                {
-                    externalVelocity = externalVelocity.Copy(y: externalVelocity.y - 1f);
-                }
+            //the only thing i don't like about this is that then there is a
```

Comment:
Also i think these comments can be added in the code review rather than the actual code i think, since otherwise we're risking having weird meaningless comments in our code down the line (not saying its meaningless now but in the future it will be lol)